sinonjs/sinon
# Support returning promises in sinon.test. Fixes #1119.

#### Purpose (TL;DR) - mandatory
Fix issue #1119 by calling thenables if present.

This is a backport of the [accepted solution in sinon-test](https://github.com/sinonjs/sinon-test/issues/6).

#### Background (Problem in detail)  - optional
The problem is fully described in #1119.

Interestingly, this only bit us when https://github.com/petkaantonov/bluebird/issues/1170 was fixed.

#### Solution  - optional
This solution is simply a backport with additional tests.

#### How to verify - mandatory
1. Check out this branch (see github instructions below)
2. `npm install`
3. `npm test`.
